roms_tools/setup/utils.py

```diff
@@ -428,6 +489,126 @@ def get_variable_metadata():
+    variable or a constant value if no related variable is available. The resulting
+    variables are added to the dictionary.
+
+    Parameters
+    ----------
+    bgc_data : dict
+        A dictionary containing biogeochemical variables as xarray DataArrays.
+        Missing variables are computed and added to this dictionary.
+
+        Assumptions:
+        - If `Fe` is part of the input dictionary, it is in units of mmol m-3.
+        - If `CHL` is part of the input dictionary, it is in units of mg m-3.
+        - If `ALK` is part of the input dictionary, it is in units of meq m-3 = mmol m-3.
+        - If `DIC` is part of the input dictionary, it is in units of mmol m-3.
+
+    Returns
+    -------
+    dict
+        The updated dictionary with missing BGC variables filled in.
+
+    Notes
+    -----
+    - If `NH4`, `DOC`, `DON`, `DOP`, `DOCr`, `DONr`, and `DOPr` are not part of the input
+      dictionary, they are filled with constant values.
+    - `CHL` is removed from the dictionary after the necessary calculations.
+    """
+
+    # Define the relationships for missing variables
+    variable_relations = {
+        "NH4": (None, 0.0),
+        "Lig": ("Fe", 3),  # is that in mmol m-3?
+        "DIC_ALT_CO2": ("DIC", 1),
+        "ALK_ALT_CO2": ("ALK", 1),
+        "DOC": (None, 0.0),
+        "DON": (None, 1.0),  # is that in mmom m-3?
+        "DOP": (None, 0.1),  # is that in mmol m-3?
+        "DOCr": (None, 10**-6),  # is that in mmol m-3?
+        "DONr": (None, 0.8),  # is that in mmol m-3?
+        "DOPr": (None, 0.003),  # is that in mmol m-3?
+        "zooC": ("CHL", 1.35),  # is that in mmol m-3?
+        "spChl": ("CHL", 0.675),  # is that in mg m-3?
+        "spC": ("CHL", 3.375),  # is that in mmol m-3?
+        "spP": ("CHL", 0.03),  # is that in mmol m-3?
+        "spFe": ("CHL", 1.35e-5),  # is that in mmol m-3?
+        "spCaCO3": ("CHL", 0.0675),  # is that in mmol m-3?
+        "diatChl": ("CHL", 0.0675),  # is that in mg m-3?
+        "diatC": ("CHL", 0.2025),  # is that in mmol m-3?
+        "diatP": ("CHL", 0.02),  # is that in mmol m-3?
+        "diatFe": ("CHL", 1.35e-6),  # is that in mmol m-3?
+        "diatSi": ("CHL", 0.0675),  # is that in mmol m-3?
+        "diazChl": ("CHL", 0.0075),  # is that in mg m-3?
+        "diazC": ("CHL", 0.0375),  # is that in mmol m-3?
+        "diazP": ("CHL", 0.01),  # is that in mmol m-3?
+        "diazFe": ("CHL", 7.5e-7),  # is that in mmol m-3?
+    }
+
+    # Fill in missing variables using the defined relationships
+    for var_name, (base_var, factor) in variable_relations.items():
+        if var_name not in bgc_data:
+            if base_var:
+                bgc_data[var_name] = bgc_data[base_var] * factor
+            else:
+                bgc_data[var_name] = factor * xr.ones_like(bgc_data["ALK"])
+
+    bgc_data.pop("CHL", None)
+
+    return bgc_data
+
+
+def compute_missing_surface_bgc_variables(bgc_data):
```

Comment:
@abigale-wyatt, could you review the code for this function as well and check for the right units?

Reply:
@abigale-wyatt did you have a look at this function as well? I think the units are wrong (see my comments in this function).